include/boost/geometry/views/enumerate_view.hpp

```diff
@@ -0,0 +1,158 @@
+// Boost.Geometry (aka GGL, Generic Geometry Library)
+
+// Copyright (c) 2024 Barend Gehrels, Amsterdam, the Netherlands.
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_GEOMETRY_VIEWS_ENUMERATE_VIEW_HPP
+#define BOOST_GEOMETRY_VIEWS_ENUMERATE_VIEW_HPP
+
+#include <boost/iterator/iterator_facade.hpp>
+#include <boost/iterator/iterator_categories.hpp>
+#include <boost/range/begin.hpp>
+#include <boost/range/end.hpp>
+#include <boost/range/difference_type.hpp>
+#include <boost/range/reference.hpp>
+#include <boost/range/value_type.hpp>
+
+#include <boost/geometry/util/type_traits_std.hpp>
+
+namespace boost { namespace geometry
+{
+
+namespace util
+{
+
+// This view is a range of values, each with an index
+// It is used to iterate over a range, and to get the index of the value
+// It is used in the enumerate function
+// The typename Range can either be const or non-const
+template <typename Range>
+struct enumerated_view
+{
+    // The return value of the iterator
+    struct value_with_index
+    {
+        using type = util::transcribe_const_t
+            <
+                Range,
+                typename boost::range_value<Range>::type
+            >;
+
+        // Member variable index contains the zero-based index of the value in the range
+        std::size_t const index;
+
+        // Member variable value contains a const or non-const reference to the value itself
+        type& value;  
+    };
+
+ private:
```

Comment:
Please use the indentation of "public:" below. Similarly adjust the comment below.

Reply:
✅